src/main/java/io/supertokens/utils/SemVer.java

```diff
@@ -0,0 +1,88 @@
+/*
+ *    Copyright (c) 2023, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ *    This software is licensed under the Apache License, Version 2.0 (the
+ *    "License") as published by the Apache Software Foundation.
+ *
+ *    You may not use this file except in compliance with the License. You may
+ *    obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *    License for the specific language governing permissions and limitations
+ *    under the License.
+ */
+
+package io.supertokens.utils;
+
+import java.util.Objects;
+
+public class SemVer implements Comparable<SemVer> {
+    public static final SemVer v2_7 = new SemVer("2.7");
+    public static final SemVer v2_8 = new SemVer("2.8");
+    public static final SemVer v2_9 = new SemVer("2.9");
+    public static final SemVer v2_10 = new SemVer("2.10");
+    public static final SemVer v2_11 = new SemVer("2.11");
+    public static final SemVer v2_12 = new SemVer("2.12");
+    public static final SemVer v2_13 = new SemVer("2.13");
+    public static final SemVer v2_14 = new SemVer("2.14");
+    public static final SemVer v2_15 = new SemVer("2.15");
+    public static final SemVer v2_16 = new SemVer("2.16");
+    public static final SemVer v2_17 = new SemVer("2.17");
+    public static final SemVer v2_18 = new SemVer("2.18");
+
+    final private String version;
+
+    public final String get() {
+        return this.version;
+    }
+
+    public SemVer(String version) {
+        if(version == null)
+            throw new IllegalArgumentException("Version can not be null");
+        if(!version.matches("[0-9]+(\\.[0-9]+)*"))
+            throw new IllegalArgumentException("Invalid version format");
+        this.version = version;
+    }
+
+    @Override public int compareTo(SemVer that) {
```

Comment:
- You should add extensive unit tests for this.
- Also, this function can be very easily misused.. it's not clear what it means if this function returns `-1`. Instead, i would prefer to have a function like isGreaterThan, isLesserThan, isEqualTo

Reply:
This is the default `Comparable` interface from java that I think we should definitely implement. I'll add those functions in the other core PR since they will be used there. I debated adding them here and replacing equality checks, but I wanted to make this PR with minimal logic changes.
Btw, we basically only need:
- `equals`
- `atLeast` (or `greaterThanOrEqualTo`)
- `betweenInclusive` (for version ranges like `2.9`-`2.18`)
with maybe a `lesserThan` (but not equal) in the future

Reply:
Yea. Please implement greaterThanOrEqualTo, betweenInclusive and lesserThan